### PR TITLE
GUI-2341 add memcached as package dependency, install config file and start/stop w/ eucaconsole

### PR DESCRIPTION
**Pull request by dkavanagh**
add memcached as package dependency, install config file and start/stop w/ eucaconsole

### CODE REVIEW

File: eucaconsole.init [hunk 1]
```diff
@@ -46,10 +46,21 @@ PROCESS=eucaconsole
 LOGFILE=/var/log/eucaconsole_startup.log
 PIDFILE=/var/run/eucaconsole/eucaconsole.pid
 NGINXPIDFILE=$(grep nginx.pid /etc/eucaconsole/nginx.conf | awk {'print $2'} | sed s'/\;//')
+MEMCACHEDPIDFILE=/var/run/memcached/eucaconsole-memcached.pid
```
Comment (gholms) [inline eucaconsole.init]: This is a separate copy of memcached, so put it in /var/run/eucaconsole instead.

File: eucaconsole.init [hunk 2]
```diff
@@ -66,8 +77,18 @@ start() {
         echo
     fi
 
+    # Ensure that /var/run/memcached has proper permissions
+    if [ "`stat -c %U /var/run/memcached`" != "$USER" ]; then
+        chown $USER /var/run/memcached
+    fi
```
Comment (gholms) [inline eucaconsole.init]: We don't own the sysadmin's copy of memcached.

Comment (gholms) [inline eucaconsole.init]: We don't own the system's copy of memcached.  If the person deploying the service goes out of his or her way to change the name of the user this runs as then it is fine to let him or her handle permissions as well.

File: eucaconsole.init [hunk 3]
```diff
@@ -46,10 +46,21 @@ PROCESS=eucaconsole
...
+MEMCACHEDPIDFILE=/var/run/memcached/eucaconsole-memcached.pid
 CONSOLEUSER=eucaconsole
 
 LOCKFILE=/var/lock/subsys/eucaconsole
 
+# memcached defaults
+USER=memcached
+MAXCONN=1024
+CACHESIZE=64
+OPTIONS=""
```
Comment (gholms) [inline eucaconsole.init]: Considering this is eucaconsole's init script, not memcached's most of these variables have nowhere near the effects that their names suggest.

File: eucaconsole.init [hunk 4]
```diff
@@ -66,8 +77,18 @@ start() {
...
+    fi
+
+    echo -n "Starting eucaconsole memcached: "
+    daemon --user=$USER --pidfile ${MEMCACHEDPIDFILE} memcached -d -p $PORT -u $USER  -m $CACHESIZE -c $MAXCONN -P ${MEMCACHEDPIDFILE} $OPTIONS
+    [ $? -eq 0 ] && touch /var/lock/subsys/eucaconsole-memcached
+    echo
+
     echo -n $"Starting $SERVICE: "
-    daemon --user=$CONSOLEUSER --pidfile=$PIDFILE $PROCESS -d -p $PIDFILE -l $LOGFILE
+    daemon --user=$CONSOLEUSER --pidfile=$PIDFILE $PROCESS -d -p $PIDFILE
```
Comment (gholms) [inline eucaconsole.init]: This seems unrelated to this pull request.

**Comment by gholms** [comment]
It would be more consistent to handle memcached the same way as nginx, namely by adding something along the lines of MEMCACHED_FLAGS to /etc/sysconfig/eucaconsole and not starting the service if one explicitly tells it not to.  The latter is particularly important when one deploys the console using a shared instance of memcached.  That would also mean that there is no need to add another configuration file.

File: eucaconsole.init [hunk 5]
```diff
@@ -60,9 +70,16 @@ start() {
...
+        echo -n "Starting eucaconsole nginx: "
         daemon /usr/sbin/nginx -c /etc/eucaconsole/nginx.conf $NGINX_FLAGS
-	[ $? -eq 0 ] && touch /var/lock/subsys/eucaconsole-nginx
+        [ $? -eq 0 ] && touch /var/lock/subsys/eucaconsole-nginx
+        echo
+    fi
+
+    if [ "x$MEMCACHED_FLAGS" != xNO ]; then
+        echo -n "Starting eucaconsole memcached: "
+        daemon --user=$CONSOLEUSER --pidfile ${MEMCACHEDPIDFILE} memcached -d -p $PORT -u $CONSOLEUSER  -m $MEMCACHED_CACHESIZE -c $MEMCACHED_MAXCONN -P ${MEMCACHEDPIDFILE} $MEMCACHED_FLAGS
```
Comment (gholms) [inline eucaconsole.init]: I haven't had a chance to test it yet, but won't -P cause memcached to try to write its own PID file?  The init script is already taking care of that.

Comment (dkavanagh) [inline eucaconsole.init]: I wondered about that too. I copied from the memcached init script. Let me double check.

Comment (gholms) [inline eucaconsole.init]: Okay.  In that case just fix the date in the changelog entry and this should be good to go.  I'll merge it tomorrow.

Thanks for the patch!
